scripts/exgfs_wave_stat_pnt.sh

```diff
@@ -0,0 +1,341 @@
+
+    for stype in mean spread prob
+	do
+    # Correct indirect variable expansion
+    	dir_var="COMOUT_WAVE_GRIB_${wavepostGRD}"
+    	cpdir="${!dir_var}"
+
+    # Ensure directory exists before proceeding
+    	if [ -z "$cpdir" ] || [ ! -d "$cpdir" ]; then
+        	echo "Error: Directory '$cpdir' does not exist or is empty."
+        exit 2
+    	fi
+
+    # Use ls to safely check for matching files
+    for file in "${cpdir}/${WAV_MOD_TAG}.t${cyc}z.${stype}.${grdNAME}.f"???.grib2; do
+      if [[ -f "$file" ]]; then  # Ensure it's a file before linking
+	      ln -s "$file" .
+        else
+              msg="ABNORMAL EXIT: Error in copying $cpfile"
+              postmsg "$msg"
+              echo ' '
+              echo '******************************************************* '
+              echo "*** FATAL ERROR: No $cpfile copied. *** "
+              echo '******************************************************* '
+              echo ' '
+              echo "$cpfile missing." >> "$ensemb_log"
+              export err=2
+              [ -n "$errchk" ] && eval "$errchk"
+              exit "$err"
+        fi
+        done	    
+      done
+
+#
+# 3. Generate bulletin and time series files at complete set of buoy locations
+#
+  
+# 3.a Buoy locations file massaging
+
+  sed '/\$/d' buoy_file.data | sed '/STOPSTRING/d ' > buoy.file
+  
+  nbuoys=`cat buoy.file | wc -l`
+
+# 3.b Command file set-up
+  rm -f cmdfile cmdfile.$ 
+
+  ibuoy=1
+
+# 3.c Create bundled grib2 file with all parameters
+
+  
+
+  cat ${WAV_MOD_TAG}.t${cyc}z.{mean,prob,spread}.${grdNAME}.f???.grib2 | $WGRIB2 - -match "(HTSGW|PERPW|WIND)" -grib gribfile > gribfile.out 2>&1 
+
+  if [ -s gribfile ]
+  then
+     set +x
+     echo "   Gribfile for bulletins created"
+     [[ "$LOUD" = YES ]] && set -x
+    else
+      set +x
+      echo ' '
+      echo '************************************************* '
+      echo "*** FATAL ERROR: No gribfile created for ${TYPE}, no bulls  *"
+      echo '************************************************* '
+      echo ' '
+      echo "$modIE fcst $date $cycle: No gribfile created." >> $wavelog
+      echo $msg
+      [[ "$LOUD" = YES ]] && set -x
+      export err=7;${errchk}
+      exit $err
+    fi
+
+
+# 3.d Loop through buoys and populate cmdfiles with calls to wave_ens_bull.sh
+  ifile=0
+  while [ ${ibuoy} -le ${nbuoys} ]
+  do
+
+    bline=`sed ''$ibuoy'!d' buoy.file`
+    blat=`echo $bline | awk '{print $2}'`
+    blon=`echo $bline | awk '{print $1}'`
+    bnom=`echo $bline | awk '{print $3}' | sed "s/'//g"`
+
+    echo "$HOMEgfs/ush/wave_ens_bull.sh ${blon} ${blat} ${bnom} 1> bull_${bnom}.out 2>&1" >> cmdfile
+
+    ibuoy=`expr ${ibuoy} + 1`
+    ifile=`expr ${ifile} + 1`
+
+  done
+
+
+
+# 3.e Execute poe or serial cmdfile
+  set +x
+  echo ' '
+  echo " Generating bulletins and ts files for ${nbuoys} locations."
+  echo ' '
+  [[ "$LOUD" = YES ]] && set -x
+
+    "${HOMEgfs}/ush/run_mpmd.sh" cmdfile    
```

Comment:
@aerorahul this is where I need to change as the cmdfile here will have 240 lines and the model crashed.